src/yunohost/utils/password.py

```diff
@@ -0,0 +1,197 @@
+
+        self.listed = unlisted < 320000000
+        return self.compare(unlisted, length, digits, lowers, uppers, others)
+
+    def compare(self, unlisted, length, digits, lowers, uppers, others):
+        strength = 0
+
+        for i, config in enumerate(self.strength_lvl):
+            if unlisted < config[0] or length < config[1] \
+               or digits < config[2] or lowers < config[3] \
+               or uppers < config[4] or others < config[5]:
+                break
+            strength = i + 1
+        return strength
+
+    def is_in_online_pwned_list(self, password, silent=True):
+        """
+        Check if a password is in the list of breached passwords from
+        haveibeenpwned.com
+        """
+
+        from hashlib import sha1
+        import requests
+        hash = sha1(password).hexdigest()
+        range = hash[:5]
+        needle = (hash[5:].upper())
+
+        try:
+            hash_list =requests.get('https://api.pwnedpasswords.com/range/' +
+                                      range, timeout=30)
+        except e:
+            if not silent:
+                raise
+        else:
+            if hash_list.find(needle) != -1:
+                return True
+        return False
+
+    def is_in_cracklib_list(self, password, pwd_dict):
+        try:
+            cracklib.VeryFascistCheck(password, None,
+                                      os.path.join(PWDDICT_PATH, pwd_dict))
+        except ValueError as e:
+            # We only want the dictionnary check of cracklib, not the is_simple
+            # test.
+            if str(e) not in ["is too simple", "is a palindrome"]:
+                return True
+
+
+class ProfilePasswordValidator(PasswordValidator):
+    def __init__(self, profile):
+        self.profile = profile
+        import json
+        try:
+            settings = json.load(open('/etc/yunohost/settings.json', "r"))
+            self.validation_strength = int(settings["security.password." + profile +
+                '.strength'])
+        except Exception as e:
+            self.validation_strength = 2 if profile == 'admin' else 1
+            return
+
+class LoggerPasswordValidator(ProfilePasswordValidator):
+    """
+    PasswordValidator class validate password
+    """
+
+    def validate(self, password):
+        """
+        Validate a password and raise error or display a warning
+        """
+        if self.validation_strength == -1:
+            return
+        import errno
+        import logging
+        from moulinette import m18n
+        from moulinette.core import MoulinetteError
+        from moulinette.utils.log import getActionLogger
+
+        logger = logging.getLogger('yunohost.utils.password')
+
+        status, msg = super(LoggerPasswordValidator, self).validate(password)
+        if status == "error":
+            raise MoulinetteError(1, m18n.n(msg))
+        elif status == "warning":
+            logger.info(m18n.n(msg))
+
+if __name__ == '__main__':
```

Comment:
Should have a comment here explaining why this is needed and why not use `yunohost tools validatepw` instead